sdk/core/Azure.Core.Experimental/src/DynamicJson.DynamicMetaObject.cs

```diff
@@ -83,6 +119,14 @@ public override DynamicMetaObject BindSetMember(SetMemberBinder binder, DynamicM
                 BindingRestrictions restrictions = BindingRestrictions.GetTypeRestriction(Expression, LimitType);
                 return new DynamicMetaObject(setCall, restrictions);
             }
+
+            private static Dictionary<Type, MethodInfo> GetCastFromOperators()
+            {
+                return typeof(DynamicJson)
+                    .GetMethods(BindingFlags.Public | BindingFlags.Static)
+                    .Where(method => method.Name == "op_Explicit" || method.Name == "op_Implicit")
```

Comment:
We should cache these MethodInfos in a static.

Reply:
I believe I am doing what you're suggesting above here: https://github.com/Azure/azure-sdk-for-net/pull/33063/files#diff-9fa0e6f4c4c212bff6545aca2d81f128cf2ac7c6f68db22ef905ecbe484ce1d7R81

But let me know if you had something else in mind ...

Reply:
If you have suggestions to refactor to make this more clear, I'm happy to take them.

Reply:
Refactored for clarity.